cmd/keycmd/list.go

```diff
@@ -646,16 +646,21 @@ func formatCChainBalance(balance *big.Int) (string, error) {
 	if useGwei {
 		return fmt.Sprintf("%d", balance), nil
 	}
-	// convert to nAvax
-	balance = balance.Div(balance, big.NewInt(int64(units.Avax)))
-	if balance.Cmp(big.NewInt(0)) == 0 {
+
+	// convert to nAvax with rounding
+	divisor := big.NewInt(int64(units.Avax))
+	half := big.NewInt(int64(units.Avax) - 1)
+	adjusted := new(big.Int).Add(balance, half)
+	result := new(big.Int).Div(adjusted, divisor)
```

Comment:
not sure why we can't just leave balance = balance.Div(balance, big.NewInt(int64(units.Avax)))

Reply:
Basic division here loses the precision. The problem is that division of big.Int does not round the numbers, which caused failure of assertions when the fee was calculated in tests. Using the logic that I added we make sure to round the number if needed which results in more accurate balance output. I can provide examples if needed